src/main/java/commission/service/SaleServiceImpl.java

```diff
@@ -28,35 +31,28 @@ public List<Sale> findAll(){
     @Override
     public void insertSale(Sale sale){
         saleDao.insertSale(sale);
+        logger.info("New sale created");
     }
 
+
     @Override
     public Optional<Sale> findSaleById(long id) {
-        var temp = saleDao.findSaleById(id);
-        Sale sale = null;
-        if (temp.isPresent()) {
-            sale = temp.get();
+        if (saleDao.findSaleById(id).isPresent()){
+            return saleDao.findSaleById(id);
```

Comment:
Double database read here, rather read it into a variable like you had before and then you can `return temp.get()` if it's present